### ISSUE TITLE
Allow expecting any Throwable

### ISSUE THREAD

**Issue by SamMousa** [Codeception/Codeception#4997]
#### What are you trying to achieve?
I'm trying to catch a `TypeError`.

#### What do you get instead?
Since `expectException()` only checks the class of anything that's a subclass of `\Exception`, throwables like `\TypeError` are not caught.

Snippet from https://github.com/Codeception/Codeception/blob/2.4/src/Codeception/Module/Asserts.php#L437
```php
try {
            $callback();
        } catch (\Exception $e) {
            if (!$e instanceof $class) {
                $this->fail(sprintf("Exception of class $class expected to be thrown, but %s caught", get_class($e)));
            }

```
I propose catching `\Throwable` and doing the check.

**Comment by DavertMik** [comment]
I think `exceptException` is very explicit and it doesn't say it expects Throwable :)

You can add send PR with 2 methods:

- exceptError (for everything which is not an exception)
- exceptThrow (for any throwable)

**Comment by Naktibalda** [comment]
expectThrowable was implemented by #5213 long time ago.
